FreeIT/src/lesson_10/Library.java

```diff
@@ -0,0 +1,84 @@
+package lesson_10;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.ListIterator;
+
+public class Library {
+
+	private ArrayList<Book> books;
+
+	public Library() {
+		super();
+			}
+
+	public Library(ArrayList<Book> books) {
+		super();
+		this.books = books;
+	}
+
+	public void addBook(Book book) {
+
+		int count = 0;
+		if (books != null) {
+			Iterator<Book> iter = books.listIterator();
+			while (iter.hasNext()) {
+				Book b = iter.next();
+				if (b.getId() == book.getId()) {
+					count++;
+				}
+			}
+			if (count == 0) {
+				books.add(book);
+			}
+		} else {
+			ArrayList<Book> b = new ArrayList<>();
+			b.add(book);
+			this.books = b;
+		}
+	}
+
+	public List<Book> getAll() {
+		return books;
+	}
+
+	public void delBook(int id) {
+		if (books != null) {
+			Iterator<Book> iter = books.listIterator();
+			while (iter.hasNext()) {
+				Book b = iter.next();
+				if (b.getId() == id) {
+					iter.remove();
+				}
+			}
+		}
+	}
+
+	public void correctBook(Book book) {
+		if (books != null) {
+			ListIterator<Book> iter = books.listIterator();
+			while (iter.hasNext()) {
+				Book b = iter.next();
+				if (b.getId() == book.getId()) {
+					iter.set(book);
```

Comment:
Every time you editing book - you creating new one. It's not correct

Reply:
No, I don't. I create an iterator, not a book. Isn't it?